File: pharmcrew-admin/src/main/java/kr/ant/kpa/pharmcrew/service/EducationService.java
package kr.ant.kpa.pharmcrew.service;

import kr.ant.kpa.pharmcrew.resp.PcResp;
import kr.ant.kpa.pharmcrew.resp.education.EduAttendListResp;
import kr.ant.kpa.pharmcrew.resp.education.EducationListResp;
import kr.ant.kpa.pharmcrew.resp.education.EducationResp;
import kr.ant.kpa.pharmcrew.validation.exception.FailSaveFileException;
import kr.ant.kpa.pharmcrew.validation.exception.gen.InvalidJsonFormatException;
import org.springframework.web.multipart.MultipartFile;

public interface EducationService {
    EducationListResp educationList(String startDate, String endDate, Long organizeId, String keyword, String isOpen, Integer page, Integer cntPerPage);
    PcResp registEducation(String authType, Long organizeId, String isOpen, String title,
                           String startDate, String endDate, String startHour, String startMinute, String endHour, String endMinute,
                           String takeHour, String takeMinute,
                           String qrType,
                           String address, String addressDetail, String longitude, String latitude,
                           String wayGuide, String wayDetail,
                           MultipartFile mapFile, MultipartFile timetableFile,
                           String master, String masterPhone,
                           String coursessJson,
                           long regId) throws FailSaveFileException, InvalidJsonFormatException;
    EducationResp education(Long educationId);
    PcResp updateEducation(Long educationId,
                           Long organizeId, String isOpen, String title,
                           String startDate, String endDate, String startHour, String startMinute, String endHour, String endMinute,
                           String takeHour, String takeMinute,
                           String address, String addressDetail, String longitude, String latitude,
                           String wayGuide, String wayDetail,
                           MultipartFile mapFile, MultipartFile timetableFile,
                           String master, String masterPhone,
                           String coursessJson,
                           Long delMapFileId, Long delTimetableFileId,
                           long modId) throws FailSaveFileException, InvalidJsonFormatException;
    PcResp deleteEducation(Long educationId);

    EduAttendListResp eduAttendList(Long educationId, Long courseId, String type, String keyword, Integer page, Integer cntPerPage);
    PcResp updateEduAttendApproval(Long educationId, Long[] attendIds, String approval);
}

File: pharmcrew-admin-ng/src/app/domain/education/education.service.ts
import { Injectable } from '@angular/core';
import { HttpService } from 'src/app/http.service';
import { EducationListReq } from './education-list-req';
import { EducationListResp } from './education-list-resp';
import { RegistEducationReq } from './regist-education-req';
import { PcResp } from 'src/app/pc-resp';
import { EducationResp } from './education-resp';
import { UpdateEducationReq } from './update-education-req';
import { EduAttendListReq } from './edu-attend-list-req';
import { EduAttendListResp } from './edu-attend-list-resp';
import { UpdateEduAttendApprovalReq } from './update-edu-attend-approval-req';

@Injectable({
  providedIn: 'root'
})
export class EducationService extends HttpService {
  registEducation(req: RegistEducationReq) {
    return this.post<PcResp>('/educations', req);
  }
  educationList(req: EducationListReq) {
    return this.get<EducationListResp>('/educations', req);
  }
  education(educationId: number) {
    return this.get<EducationResp>(`/educations/${educationId}`);
  }
  updateEducation(educationId: number, req: UpdateEducationReq) {
    return this.post<PcResp>(`/educations/${educationId}`, req);
  }
  deleteEducation(educationId: number) {
    return this.delete<PcResp>(`/educations/${educationId}`);
  }

  eduAttendList(educationId: number, req: EduAttendListReq) {
    return this.get<EduAttendListResp>(`/educations/${educationId}/attends`, req);
  }
  updateEduAttendApproval(educationId: number, req: UpdateEduAttendApprovalReq) {
    return this.put<PcResp>(`/educations/${educationId}/attendapproval`, req);
  }
}
